.github/workflows/ci.yaml

```diff
@@ -0,0 +1,50 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - celo*
+
+  pull_request:
+    branches:
+      - master
+      - celo*
+
+jobs:
+  Test:
+    runs-on: ubuntu-latest
+    steps:
+        - uses: actions/checkout@v4
+
+        - uses: actions/setup-go@v4
+          with:
+            go-version: '1.19'
+
+        - name: Build
+          run: make all
+
+        - name: Run e2e tests
+          shell: bash
+          run: |
+            curl -L https://foundry.paradigm.xyz | bash
+            source /home/runner/.bashrc
+            PATH="~/.config/.foundry/bin:${PATH}"
+            foundryup
+            e2e_test/run_all_tests.sh
+
+        - name: Test
+          run: make test
+
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+        - uses: actions/checkout@v4
+
+        - uses: actions/setup-go@v4
+          with:
+            go-version: '1.19'
+            cache: false
```

Comment:
Why not cache it?

Reply:
The problem is that both the test and the lint jobs would use the same cache hash. As as the test job runs longer, it would never be able to write the cache with its bigger cache.

We can also solve this by doing caching manually, but as the lint job is so short, I didn't see a problem with disabling it here.